common/djangoapps/custom_settings/views.py

```diff
@@ -0,0 +1,64 @@
+import json
+import logging
+
+import django.utils
+from django.contrib.auth.decorators import login_required
+from django.core.exceptions import ObjectDoesNotExist
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseBadRequest, HttpResponseNotAllowed
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.decorators.http import require_http_methods
+from opaque_keys.edx.keys import CourseKey
+
+from cms.djangoapps.contentstore.views.course import get_course_and_check_access
+from edxmako.shortcuts import render_to_response
+from util.json_request import JsonResponse, expect_json
+from xmodule.modulestore.django import modulestore
+from .models import CustomSettings
+
+log = logging.getLogger(__name__)
+
+
+@login_required
+@ensure_csrf_cookie
+@require_http_methods(("GET", "POST", "PUT"))
+@expect_json
+def custom_settings(request, course_key_string):
+    """
+    Course custom settings configuration
+    GET
+        html: get the page
+    PUT, POST
+        json: update the Course's custom settings.
+    """
+
+    course_key = CourseKey.from_string(course_key_string)
+
+    with modulestore().bulk_operations(course_key):
+        course_module = get_course_and_check_access(course_key, request.user)
+        try:
+            settings = CustomSettings.objects.get(course_key=course_key)
+
+        except ObjectDoesNotExist as exc:
+
+            return HttpResponseBadRequest(django.utils.html.escape(exc.message), content_type="text/plain")
+
+        if 'text/html' in request.META.get('HTTP_ACCEPT', '') and request.method == 'GET':
+
+            return render_to_response('custom_settings.html', {
```

Comment:
Where is `custom_settings.html` and what is its use case?

Reply:
It is in the themes **philu-edx-theme** directory.
https://github.com/arbisoft/philu-edx-theme/pull/20